src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -386,9 +386,20 @@ public ClusterState execute(final ClusterState currentState) throws Exception {
                                 if (mergeResult.hasConflicts()) {
                                     throw new MergeMappingException(mergeResult.buildConflicts());
                                 }
+                            } else {
+                                // TODO: can we find a better place for this validation?
```

Comment:
Why not do this in the mapper service itself, when the type is added? Yes it would happen on every node the index exists when mappings are pushed there, but this is a very minimal check, and I would rather the mapper service be self sufficient than rely on something external to ensure its consistency?